src/utils/DepositUtils.ts

```diff
@@ -71,3 +73,38 @@ export async function queryHistoricalDepositForFill(
 
   return validateFillForDeposit(fill, deposit) ? deposit : undefined;
 }
+
+export function createDepositForSimulatingGas(
```

Comment:
Do we create a deposit just to pass to the fill creation function? Any reason we can't just create the fill directly and avoid creating the intermediate deposit struct?

Reply:
I mainly created this extra step so that we can re-use the gas estimation step for an arbitrary fill & so that we could construct an arbitrary deposit. I imagine this may have further usage, but we can definitely close the loop and revisit encapsulating logic in a different PR

Reply:
> we can re-use the gas estimation step for an arbitrary fill & so that we could construct an arbitrary deposi

What's the use case for this, out of curiosity? I also thought it was kind of strange to create a deposit to create a fill, rather i'd stick to constructing a fill directly

Reply:
This structure is like this because the `QueryInterface` where that abstracts the various chains needs a common data-structure as input. The `ProfitClient` and  Vercel API references deposits so if possible, I will change the `getGasCosts` to reference a `Deposit` instead of a fill.

Reply:
Done: https://github.com/across-protocol/sdk-v2/pull/427/commits/b94563648e64e6a17bd4cb3f89da378cd959a1d6

Ultimately, I was originally trying to make a clean interface and the closest way I thought to do this robustly was to convert `Parameters -> Deposit`, then `Deposit -> Fill`, then simulate that fill. However, we could stop this and only require the values needed so `Parameters -> Fill`